0005-crashtracker-structured-log-format.md

```diff
@@ -0,0 +1,203 @@
+- `incomplete`: **[required]**
+  Boolean `false` if the crashreport is complete (i.e. contains all intended data), `true` if there is expected missing data.
+  This can happen becasue the crashtracker is architected to stream data to an out of process receiver, allowing a partial crash report to be emitted even in the case where the crashtracker itself crashed during stack trace collection.
+  This MUST be set to `true` if any required field is missing.
+- `metadata`: **[required]**
+  Metadata about the system in which the crash occurred:
+  - `library_name`: **[required]**
+    e.g. "dd-trace-python".
+  - `library_version`: **[required]**
+    e.g. "2.16.0".
+  - `family`: **[required]**
+    e.g. "python".
+  - `tags`: **[optional]**
+    A set of key:value pairs, representing any tags the crashtracking system wishes to associate with the crash.
+    Examples would include "hostname", "service", and any configuration information the system wishes to track.
+- `os_info`: **[required]**
+  The OS + processor architecture on which the crash occurred. TODO: define how these should look with examples.
+  - `architecture`: **[required]**
+  - `bitness`: **[required]**
+  - `os_type`: **[required]**
+  - `version`: **[required]**
+- `proc_info`: **[optional]**
+  A place to store information about the crashing process.
+  In the future, this may have additional optional fields as more data is collected.
+  - `pid`: **[required]**
+    The PID of the crashing process.
+- `sig_info`: **[optional]**
+  UNIX signal based collectors only: Useful information from the [siginfo_t](https://man7.org/linux/man-pages/man2/sigaction.2.html) structure.
+  - `faulting_address`: **[optional]**
+    A hexidecimal string with the memory address at which the fault occurred, e.g. "0xdeadbeef".
+  - `signame`: **[required]**
+    The signal name, e.g. "SIGSEGV".
+  - `signum`: **[required]**
+    An integer storing the [UNIX signal number](https://man7.org/linux/man-pages/man7/signal.7.html), e.g. `11` for a segmentation violation.
+- `span_ids`: **[optional]**
+  A vector representing active span ids at the time of program crash.
+  The collector MAY cap the number of spans that it tracks.
+  - `span_id`: **[required]**
+    A string containing the span id.
+  - `thread_id`: **[optional]**
+    A string containing the thread_id for the given span.
+- `timestamp`: **[required]**
+  The time at which the crash occurred, in ISO 8601 format.
+- `trace_ids:`: **[optional]**
+  A vector representing active span ids at the time of program crash.
+  The collector MAY cap the number of spans that it tracks.
+  - `thread_id`: **[optional]**
+    A string containing the thread_id for the given trace.
+  - `trace_id`: **[required]**
+    A string containing the trace id.
+- `uuid`: **[required]**
+  A UUID v4 which uniquely identifies the crash.
+  This will typically be generated at crash-time, and then associated with the uploaded crash.
```

Comment:
After this, should add a top-level `fingerprint` field which is an arbitrary string used by the system to fingerprint crashes.

Reply:
done